src/test/java/org/cloudfoundry/promregator/endpoint/DiscoveryEndpointControllerTest.kt

```diff
@@ -0,0 +1,65 @@
+package org.cloudfoundry.promregator.endpoint
+
+import com.ninjasquad.springmockk.MockkBean
+import io.mockk.every
+import org.cloudfoundry.promregator.config.PromregatorConfiguration
+import org.cloudfoundry.promregator.discovery.CFMultiDiscoverer
+import org.cloudfoundry.promregator.scanner.Instance
+import org.cloudfoundry.promregator.scanner.ResolvedTarget
+import org.hamcrest.CoreMatchers.`is`
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.get
+
+@ExtendWith(SpringExtension::class)
+@WebMvcTest(DiscoveryEndpoint::class, properties = ["promregator.simulation.enabled=true", "promregator.lifecycle.enabled=false"])
+class DiscoveryEndpointControllerTest(
+       @Autowired private val mockMvc: MockMvc
+) {
+    @MockkBean private lateinit var discoverer: CFMultiDiscoverer
+    @MockkBean(relaxed = true) private lateinit var config: PromregatorConfiguration
+
+    @BeforeEach
+    fun setup() {
+
+    }
+
+    /** Test for [DiscoveryEndpoint.getDiscovery] */
+    @Test
+    fun `get returns 503 when no targets are resolvable`() {
+        every {discoverer.discover(any(), any()) } returns listOf()
+
+        mockMvc.get("/discovery").andDo { print() }.andExpect {
+            status { isServiceUnavailable }
+        }
+    }
+
+    /** Test for [DiscoveryEndpoint.getDiscovery] */
+    @Test
+    fun `get returns json that includes the apps space, instance, and metrics path`() {
```

Comment:
Kotlin functions support spaces when using backticks. Not useful in app code, but is quite useful in tests